tests/attributions/test_opti_cam.py

```diff
@@ -0,0 +1,110 @@
+import pytest
+
+import tensorflow as tf
+from tensorflow.keras.models import Model
+from tensorflow.keras.layers import Dense, Conv2D, Activation, Dropout, Flatten, MaxPooling2D, Input
+
+from xplique.attributions import OptiCAM
+from xplique.attributions.opti_cam import _normalization_dict, _loss_dict
+from ..utils import generate_data, almost_equal
+
+
+def _generate_model(input_shape=(32, 32, 3), output_shape=10):
+    inputs = Input(shape=input_shape)
+    x = Conv2D(4, kernel_size=(2, 2), activation='relu', name='conv2d')(inputs)
+    x = Conv2D(4, kernel_size=(2, 2), activation='relu', name='conv2d_1')(x)
+    x = MaxPooling2D(pool_size=(2, 2))(x)
+    x = Dropout(0.25)(x)
+    x = Flatten()(x)
+    x = Dense(output_shape)(x)
+    outputs = Activation('softmax')(x)
+
+    model = Model(inputs=inputs, outputs=outputs)
+    model.compile(loss='categorical_crossentropy', optimizer='sgd')
+
+    return model
+
+
+def test_initiation():
+    """Test the initiation of the OptiCAM class"""
+    model = _generate_model()
+
+    with pytest.raises(ValueError):
+        OptiCAM(model, normalization='invalid')
+
+    with pytest.raises(ValueError):
+        OptiCAM(model, loss_type='invalid')
+
+    with pytest.raises(AssertionError):
+        OptiCAM(model, n_iters=0)
+
+
+def test_output_shape():
+    """The output shape must be the same as the input shape"""
+
+    input_shapes = [(28, 28, 1), (32, 32, 3)]
+    nb_labels = 10
+
+    for input_shape in input_shapes:
+        samples, labels = generate_data(input_shape, nb_labels, 100)
+        model = _generate_model(input_shape, nb_labels)
+
+        method = OptiCAM(model, -2)
+        outputs = method.explain(samples, labels)
+
+        assert samples.shape[:3] == outputs.shape[:3]
```

Comment:
You should also check that the outputs shape finish by a 1. There is a test for all methods where you should add OptiCAM. (tests/attributions/test_commons.py/test_data_types_shapes()).